assets/js/EditCells.js

```diff
@@ -0,0 +1,104 @@
+// Code required to add a cell edit.
+// Including creating an Edit Cells module if needed, and syncing to server
+import React from 'react'
+import {store, changeSelectedWfModuleAction} from "./workflow-reducer";
+import {getPageID} from "./utils";
+import WorkbenchAPI from './WorkbenchAPI'
+
+var api = WorkbenchAPI(); // var so it can be mocked for testing
+export function mockAPI(mock_api) {
+  api = mock_api;
+}
+
+function findParamValByIdName(wfm, paramValIdName) {
+  return wfm.parameter_vals.find((parameterVal) => {
+        return parameterVal.parameter_spec.id_name === paramValIdName;
+    });
+}
+
+// Returns wfm object index in stack,  given its global ID
+function getWfModuleIndexfromId(state, id) {
+  var wfModuleIdx = null;
+  state.workflow.wf_modules.find((wfm, idx) => {
+    wfModuleIdx = idx;
+    return wfm.id === id;
+  });
+
+  return wfModuleIdx;
+}
+
+// Look for an existing Edit Cells module at or after the edited module
+// Returns module index, null if none
+function findEditCellsModule(state, wfModuleId) {
+  var wfModules = state.workflow.wf_modules;
+  var idx = getWfModuleIndexfromId(state, wfModuleId);
+
+  // Is this an existing Edit Cells module?
+  if (wfModules[idx].module_version.module.id_name === 'editcells' ) {
+    return wfModules[idx];
+  }
+
+  // Is the next module Edit Cells? If so, we can merge this edit in
+  var nextIdx = idx + 1;
+  if (nextIdx === wfModules.length) {
+    return null;   // end of stack
+  } else if (wfModules[nextIdx].module_version.module.id_name === 'editcells' ) {
+    return wfModules[nextIdx];
+  }
+
+  // Nope, no Edit Cells where we need it
+  return null;
+}
+
```

Comment:
Everything below this comment needs to be moved in to the reducer. Leave a comment here to that effect